### PR TITLE
Add new Dockerfile with npm-token authen and CI script

### PR DESCRIPTION
**Pull request by harris1111**
## Type

- [ ] Feature
- [x] Bug fix
- [x] CI/CD
- [ ] Documentation
- [ ] Other

## Description

Add new Dockerfile for mounting npm-token as secret when building images
Add new github action script for CI.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 **/node_modules
 .git
-.github
\ No newline at end of file
+.github
+.env
```
Comment (dqtkien) [inline .dockerignore:4]: Add .npmrc too

File: .github/workflows/build-and-push-docker.yml [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  DOCKER_REPO_BROKER_SERVICE: orochinetwork/zkdb-broker-service
+  DOCKER_REPO_PROOF_SERVICE: orochinetwork/zkdb-proof-service
+  DOCKER_REPO_SERVERLESS: orochinetwork/zkdb-serverless
```
Comment (dqtkien) [inline .github/workflows/build-and-push-docker.yml:11]: lack of `NPM_TOKEN: ${{ secrets.NPM_ACCESS_TOKEN }}`

File: packages/ftp/Dockerfile [hunk 3]
```diff
@@ -15,12 +15,18 @@ USER appuser
 
 # Copy the application code as the non-root user
 COPY --chown=appuser:appuser . .
+USER root
```
Comment (dqtkien) [inline packages/ftp/Dockerfile:18]: Why we need user root here?

File: packages/serverless/Dockerfile [hunk 4]
```diff
@@ -13,17 +13,22 @@ RUN chown -R appuser:appuser /usr/src/app
...
+RUN yarn workspace @zkdb/storage build && \
+    yarn workspace @zkdb/serverless build
+
+# Set the working directory for the broker service
 WORKDIR /usr/src/app/packages/serverless
 
 # Set the entrypoint to the application
-ENTRYPOINT ["node", "build/src/index.js"]
\ No newline at end of file
+ENTRYPOINT ["node", "build/src/index.js"]
```
Comment (dqtkien) [inline packages/serverless/Dockerfile:34]: Each services have their `npm run start` command, please use it first.

File: packages/ftp/Dockerfile (hunk 3 above)
Comment (harris1111) [reply packages/ftp/Dockerfile:18]: I Think we can use the user `root` to mount the secret to /root/.npmrc, and then run `yarn install`. After that, I change to user appuser to build, so if someone hijack our image he can't see the secret file npmrc which is mount by root (also, when mounting as secret, the file is empty after the image building is done).

File: packages/serverless/Dockerfile (hunk 4 above)
Comment (harris1111) [reply packages/serverless/Dockerfile:34]: As I'm checking at the latest commit at `main` branch, there's no `start` script in `package.json`. Can you check it again?

File: .github/workflows/build-and-push-docker.yml (hunk 2 above)
Comment (harris1111) [reply .github/workflows/build-and-push-docker.yml:11]: Okay I will change the step to assign npm token to this section, Currently i'm putting it at `build` step

File: packages/serverless/Dockerfile (hunk 4 above)
Comment (dqtkien) [reply packages/serverless/Dockerfile:34]: Yeah. Use it first, if it doesn't have `start` command, then use your `node` command

File: packages/ftp/Dockerfile (hunk 3 above)
Comment (dqtkien) [reply packages/ftp/Dockerfile:18]: lgtm

File: .gitignore [hunk 5]
```diff
@@ -112,4 +112,7 @@ circuit-cache/
 .vscode/
 
 ./downloads
-./ftp-root
\ No newline at end of file
+./ftp-root
+
+#private
+npm-token
```
Comment (dqtkien) [inline .gitignore:118]: .npmrc
```suggestion
.npmrc
```

File: .gitignore [hunk 6]
```diff
@@ -112,4 +112,7 @@ circuit-cache/
 .vscode/
 
 ./downloads
-./ftp-root
\ No newline at end of file
+./ftp-root
+
```
Comment (dqtkien) [inline .gitignore:116]:
```suggestion
.npmrc
```

**Review by dqtkien — APPROVED** [review]

**Pull request merged**